exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,189 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class DominoesTest {
+	
+	@Test
+	public void emtpyInputEmptyOutputTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		ArrayList<Domino> dominoesArray = new ArrayList<>();
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertNull(chain);
```

Comment:
These tests don't appear to match up exactly with the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/dominoes/canonical-data.json). The first few tests should just check a `canChain` method which should return a boolean 🙂

Reply:
In the canonical data file it says there are two levels for this exercise. Level 1 is to determine if a chain exists and level 2 is to find a possible chain if one exists. I guess I was trying to implement level 2.

Reply:
I see, I didn't look at the canonical data carefully enough! Thanks for explaining 🙂 In which case what you're doing is fine 🙂